apps/cms-v2/src/modules/providers/payloadcms/payloadcms-client.ts

```diff
@@ -0,0 +1,170 @@
+import { createLogger } from "@saleor/apps-shared";
+import { WebhookProductVariantFragment } from "../../../../generated/graphql";
+
+import { PayloadCmsProviderConfig } from "@/modules/configuration/schemas/payloadcms-provider.schema";
+import { FieldsMapper } from "../fields-mapper";
+
+import qs from "qs";
+import { z } from "zod";
+
+type Context = {
+  configuration: PayloadCmsProviderConfig.FullShape;
+  variant: WebhookProductVariantFragment;
+};
+
+export class PayloadCMSClient {
+  private logger = createLogger({ name: "PayloadCMSClient" });
+
+  private mapVariantToPayloadFields({ configuration, variant }: Context) {
+    const fields = FieldsMapper.mapProductVariantToConfigurationFields({
+      variant,
+      configMapping: configuration.productVariantFieldsMapping,
+    });
+
+    return fields;
+  }
+
+  private constructCollectionUrl(config: PayloadCmsProviderConfig.FullShape) {
+    return `${config.payloadApiUrl}/${config.collectionName}`;
+  }
+
+  getItemsBySaleorVariantId(context: Context) {
+    const queryString = qs.stringify(
+      {
+        where: {
+          [context.configuration.productVariantFieldsMapping.variantId]: {
+            equals: context.variant.id,
+          },
+        },
+      },
+      {
+        addQueryPrefix: true,
+      },
```

Comment:
**suggestion (non-blocking)**:

We should add a comment about where this structure is coming from.

Reply:
added to the top of the class